### DISCUSSION TITLE
Option to distribute work across multiple hosts

### DISCUSSION THREAD

**Discussion by WanzenBug** [docker/buildx#2480] [General]
We recently started migrating our container image build process to use `buildx bake`. So far it has been working great, with only minor issues. In our setup there is a single "bake" invocation, which eventually results in hundreds (slightly different) container images.

We wanted to speed up the build process and were wondering if there is a way to configure buildx so that it does not always pick the first node that satisfies some platform requirements. It seems that currently if we want to build for `linux/amd64`, it will always pick the first node in the configuration (we are using a `remote` driver with multiple nodes exposing buildkit via TCP). 

I understand that this is needed in some cases when targets share some build stage, but in our case we have completely different targets defined in our bake file. Is there a way to achieve this? If not, would this be a feature that would be considered for inclusion? If so, I could start working on it.

**Comment by crazy-max** [comment]
We don't have any load balancing pattern with nodes, they are only used to match the target platform you want to build on. So even if you have 5 nodes supporting `linux/amd64` platform for a builder, only the first node will be used. One way to scale would be to switch to the `kubernetes` driver as shown in: https://docs.docker.com/build/drivers/kubernetes/#scaling-buildkit

> We recently started migrating our container image build process to use `buildx bake`. So far it has been working great, with only minor issues.

And thanks for using `bake` :tada:

**Reply by WanzenBug** [reply]
Thanks for the quick response. 

> One way to scale would be to switch to the kubernetes driver as shown.

That would probably be a bit too much effort for us, our builders are currently random collection of old hardware we do not need for other purposes :smile: 

I see [this quote](https://github.com/moby/buildkit?tab=readme-ov-file#load-balancing) from the buildlkit page:

> buildctl build can be called against randomly load balanced buildkitd daemons.

I'm wondering if this also applies to `buildx bake` and if that would work in our case? Or is there only a single connection created for one command invocation? :thinking:
